src/js/dom-scripts.js

```diff
@@ -52,17 +55,21 @@ document.addEventListener('keydown', (e)=> {
   switch(e.keyCode) {
   case 37:
     lastKeyPressed = 'left';
+    currentState.pacman.activeDirection = lastKeyPressed;
```

Comment:
You're modifying the state from outside the `crunchState` function. While this is very convenient, as the application grows is really hard to figure out who is changing what. As a rule of thumb, only the function that deals with the state, can change the state. That's why we send messages to it.